src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs

```diff
@@ -101,29 +100,49 @@ internal void RegisterMessageTypesFoundIn(IList<Type> availableTypes)
         {
             foreach (var type in availableTypes)
             {
-                RegisterMessageType(type);
+                if (isMessageType(type))
+                {
+                    RegisterMessageType(type);
+                }
+
+                foreach (var messageType in GetHandledTypes(type))
```

Comment:
could we wrap this within an `else` block as there shouldn't be any handler which is also a message type?

Reply:
Seems reasonable to me.

Reply:
Done (went with a `continue`)